# Type: feat
# Subtype: Ex
# English Summary: Added a new method declareVariable()
# Chinese Summary: 添加了一个新的方法 declareVariable()
# Entry Number: 18

# === Git Diff Content ===

diff --git a/presto-main/src/main/java/com/facebook/presto/byteCode/Scope.java b/presto-main/src/main/java/com/facebook/presto/byteCode/Scope.java
index 0b78b84..c1d4d0c 100644
--- a/presto-main/src/main/java/com/facebook/presto/byteCode/Scope.java
+++ b/presto-main/src/main/java/com/facebook/presto/byteCode/Scope.java
@@ -13,6 +13,7 @@
  */
 package com.facebook.presto.byteCode;
 
+import com.facebook.presto.byteCode.expression.ByteCodeExpression;
 import com.google.common.collect.ImmutableList;
 import org.objectweb.asm.Type;
 
@@ -102,4 +103,11 @@ public class Scope
 
         return variable;
     }
+
+    public Variable declareVariable(String variableName, ByteCodeBlock block, ByteCodeExpression initialValue)
+    {
+        Variable variable = declareVariable(initialValue.getType(), variableName);
+        block.append(variable.set(initialValue));
+        return variable;
+    }
 }
